test/evp_test.c

```diff
@@ -294,452 +206,112 @@ static int evp_test_buffer_do(STACK_OF(EVP_TEST_BUFFER) *sk,
     return 1;
 }
 
-#ifndef OPENSSL_NO_SCRYPT
-/* Currently only used by scrypt tests */
-/* Parse unsigned decimal 64 bit integer value */
-static int test_uint64(const char *value, uint64_t *pr)
-{
-    const char *p = value;
-
-    if (!TEST_true(*p)) {
-        TEST_info("Invalid empty integer value");
-        return -1;
-    }
-    *pr = 0;
-    while (*p) {
-        if (*pr > UINT64_MAX / 10) {
-            TEST_error("Integer overflow in string %s", value);
-            return -1;
-        }
-        *pr *= 10;
-        if (!TEST_true(isdigit(*p))) {
-            TEST_error("Invalid character in string %s", value);
-            return -1;
-        }
-        *pr += *p - '0';
-        p++;
-    }
-    return 1;
-}
-
 static int compare_mem(unsigned char *expected, size_t expected_len,
                        unsigned char *got, size_t  got_len)
 {
     if (!TEST_mem_eq(expected, expected_len, got, got_len))
         return 0;
     return 1;
 }
```

Comment:
Note "probably" in the *side note*. It means it's just an idea that might be worth considering. It might be sensible or it might not. But I can't prevent you from testing and telling how it looks. For example with regard to expected appearing in printout. It might be important, but then it might be possible to solve it by renaming some structure fields... Once again, something to consider...